app/models/topic.rb

```diff
@@ -126,6 +126,11 @@ def assign(user_id=2, assigned_to)
     self.save
   end
 
+  def self.bulk_assign(post_attributes, assigned_to)
+    Post.create(post_attributes)
+    self.update_all(assigned_user_id: assigned_to, current_status: 'pending')
```

Comment:
I initially panicked that `update_all` would affect _all_ topics, but see you've called this on `@topics` specifically, so it will only apply to those.